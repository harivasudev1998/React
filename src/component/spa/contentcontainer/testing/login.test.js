import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom'
import {mount} from 'enzyme';
import {shallow} from 'enzyme';
import Login from '../login/login';
import {configure} from 'enzyme'
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter:new Adapter()});

it('Render',()=>{
  const wrapper=shallow(<Login />);
  const h=wrapper.find("h1");
  const result=h.text();

  expect(result).toBe("Login");
})

test('Login', () => {
    expect(true).toBeTruthy();
  });

it('Submit Function',()=>{
  const onSubmitFun=jest.fn();
  const wrapper=mount(<form onSubmit={onSubmitFun}></form>)
  const form=wrapper.find('form');
  form.simulate('submit');
  expect(onSubmitFun).toHaveBeenCalledTimes(1);
});
it('username field',()=>{
  const wrapper=mount(<input type="text" name="username"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('text');
  expect(input.prop('name')).toEqual('username');
})
it('Password field',()=>{
  const wrapper=mount(<input type="password" name="password"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('password');
  expect(input.prop('name')).toEqual('password');
})
it('Button field',()=>{
  const wrapper=mount(<button type="submit">Login</button>);
  const input=wrapper.find('button');
  const result=input.text();
  expect(result).toBe('Login')
  expect(input.prop('type')).toEqual('submit');
})
test('matches Snapshot',()=>{
  const tree=renderer.create(<Router><Login /></Router>).toJSON()
  expect(tree).toMatchSnapshot();
})