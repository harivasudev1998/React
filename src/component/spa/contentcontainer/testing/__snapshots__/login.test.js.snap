// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="contain"
    style={
      Object {
        "backgroundColor": "lightgoldenrodyellow",
        "paddingBottom": "330px",
        "paddingTop": "10px",
      }
    }
  >
    <h1
      style={
        Object {
          "color": "blue",
        }
      }
    >
      Login
    </h1>
    <form>
      <b>
        <label>
          Username:  
        </label>
      </b>
      <input
        id="username"
        onChange={[Function]}
        type="text"
      />
      <span
        className="error"
        name="username"
      >
        Username is required
      </span>
      <br />
      <br />
      <b>
        <label>
          Password:  
        </label>
      </b>
      <input
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
      />
      <span
        className="error"
      >
        Password is required
      </span>
      <br />
      <br />
      <input
        disabled={true}
        onClick={[Function]}
        type="button"
        value="Login"
      />
      <br />
      <br />
    </form>
  </div>
</div>
`;
